### ISSUE TITLE
Potential GPL license violation

### ISSUE THREAD

**Issue by tlhunter** [godmodelabs/flora-sql-parser#29]
I noticed this project has an MIT license:
https://github.com/godmodelabs/flora-sql-parser/blob/master/LICENSE

However, it is "based on the SQL parser extracted from Alibaba's nquery module". The nquery module uses the GPLv2 license:

https://github.com/alibaba/nquery/blob/master/package.json#L34

That would mean this project may contain GPL code and could be mislabeled with the MIT license.

I'm curious, how the SQL parser was extracted from nquery?

**Comment by tlhunter** [comment]
More specifically, how is the `sql.pegjs` file generated? It's similar to the `nquery.pegjs` file used in nquery:

https://github.com/godmodelabs/flora-sql-parser/blob/daa6a1d1b8fcf6b60e4050a619548034fbd88eab/sql.pegjs#L154-L162
https://github.com/alibaba/nquery/blob/16f848553313f24da17118bea28c45b346e6b2f0/peg/nquery.pegjs#L150-L158

Was it generated independently using the peg.js tool using some sort of non-GPL source? Or is it truly extracted from the GPL'd nquery project?

**Comment by not-implemented** [comment]
Ok, you are right, thanks for the hint. We used the same template when creating all flora-packages and used MIT license everywhere.

We changed the license of flora-sql-parser to "GPL-2.0" now as a quick fix, to be compatible with nquery license.

When there is time, we would start a rewrite from scratch of the extracted components, which may be a good idea anyway for supporting different dialects of SQL.
